tests/v3_api/test_logging.py

```diff
@@ -0,0 +1,129 @@
+    "kafka-topics-ui.enabled": "false",
+    "kafka-topics-ui.ingress.enabled": "false"}
+
+CATTLE_ClUSTER_LOGGING_KAFKA_TEST = \
+    CATTLE_TEST_URL + "/v3/clusterloggings?action=test"
+CATTLE_PROJECT_LOGGING_KAFKA_TEST = \
+    CATTLE_TEST_URL + "/v3/projectloggings?action=test"
+KAFKA_CATALOG_ID = "catalog://?catalog=library&template=kafka&version=0.7.3"
+
+
+def test_send_log_to_kafka(setup_app):
+    cluster = namespace["cluster"]
+    project = namespace["project"]
+
+    valid_endpoint = "http://" + namespace["name_prefix"] + "-kafka" + \
+        "." + namespace["ns"].name + ".svc.cluster.local:9092"
+    print("kafka endpoint:" + valid_endpoint)
+    send_log_to_kafka(CATTLE_ClUSTER_LOGGING_KAFKA_TEST, valid_endpoint,
+                      cluster.id, project.id, ADMIN_TOKEN)
+    send_log_to_kafka(CATTLE_PROJECT_LOGGING_KAFKA_TEST, valid_endpoint,
+                      cluster.id, project.id, ADMIN_TOKEN)
+
+    unreachable_endpoint = "http://kafka.com:9092"
+    print("kafka endpoint:" + unreachable_endpoint)
+    send_log_to_kafka(CATTLE_ClUSTER_LOGGING_KAFKA_TEST, unreachable_endpoint,
+                      cluster.id, project.id, ADMIN_TOKEN, expected_status=500)
+    send_log_to_kafka(CATTLE_PROJECT_LOGGING_KAFKA_TEST, unreachable_endpoint,
+                      cluster.id, project.id, ADMIN_TOKEN, expected_status=500)
+
+    bad_format_endpoint = "kafka.com:9092"
+    print("kafka endpoint:" + bad_format_endpoint)
+    send_log_to_kafka(CATTLE_ClUSTER_LOGGING_KAFKA_TEST, bad_format_endpoint,
+                      cluster.id, project.id, ADMIN_TOKEN, expected_status=500)
+    send_log_to_kafka(CATTLE_PROJECT_LOGGING_KAFKA_TEST, bad_format_endpoint,
+                      cluster.id, project.id, ADMIN_TOKEN, expected_status=500)
+
+
+def send_log_to_kafka(url, endpoint, clusterId, projectId, token,
+                      expected_status=204):
+    headers = {'Authorization': 'Bearer ' + token}
+    kafkaConfig = {
+        "brokerEndpoints": [endpoint],
+        "topic": "test",
+    }
+
+    r = requests.post(url,
+                      json={"kafkaConfig": kafkaConfig,
+                            "clusterId": clusterId,
+                            "projectId": projectId},
+                      verify=False, headers=headers)
+    if len(r.content) != 0:
+        print(r.content)
+    assert r.status_code == expected_status
+
+
+@pytest.fixture(scope='module')
+def setup_app(request):
+    start = time.time()
+    timeout = 120
+    interval = 0.5
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    name = random_test_name("kafka")
+    namespace["name_prefix"] = name
+    p_client.create_app(name=name,
+                        answers=answers,
+                        targetNamespace=ns.name,
+                        externalId=KAFKA_CATALOG_ID,
+                        namespaceId=ns.id)
+    kafka_name = "statefulset:" + ns.name + ":" + name + "-kafka"
+    while True:
```

Comment:
I think we can wait for the app's state directly, check if `wait_for_condition` in common.py can be reused here

Reply:
the app condition state is for creating the app, execute helm etc, it dosesn't check whether the pod is deployed success.

Reply:
We have a `Deployed` condition which is set when all the pods are running, I think the active state of the app reflects that pods are deployed and running, am I missing something?